CleverCloud/clever-components
# chore(storybook): switch to static Clever App

Fixes #1280

## What does this PR do?

- Removes the `node-static` dependency,
- Removes the `start` npm script,
- Adds `.htaccess` to be used by the static Clever App (Apache server).

## How to review?

- Check the code,
- The preview shouldn't be impacted,
- Run the `npm run storybook:build` locally and check that the `storybook-static` directory contains a `.htaccess`.

## Things to discuss

To make Apache handle the SPA correctly & specify cache rules, we need a `.htaccess` file.
This `.htaccess` file needs to be placed at the root of the directory served by the server (`CC_WEBROOT`) which is `storybook-static`.

We cannot simply put the `.htaccess` in `storybook-static` ourselves because this directory is removed & created by the storybook build command (+ it's git ignored anyway).
We have two options:
- make the build command copy the `.htaccess` file in `storybook-static` (the option I chose in this PR),
- or add a `CC_POST_BUILD_HOOK` to move the `.htaccess` file in the `storybook-static` directory only when building on Clever Cloud.

The option I've chosen has 1 advantage:  you cannot forget to add this step, it's part of the build process.
It also has 1 disadvantage: we send `.htaccess` files even for previews which is useless because they are served by Cellar.
